photon-api/src/main/scala/com/linkedin/photon/ml/data/RandomEffectDataset.scala

```diff
@@ -392,12 +396,26 @@ object RandomEffectDataset {
       .features
       .length
 
-    keyedGameDataset
+    val dataProjectors = keyedGameDataset
       .mapValues { case (_, labeledPoint) =>
         VectorUtils.getActiveIndices(labeledPoint.features)
       }
       .foldByKey(mutable.Set[Int](), randomEffectPartitioner)(_.union(_))
       .mapValues(activeIndices => new LinearSubspaceProjector(activeIndices.toSet, originalSpaceDimension))
+
+    val sc = dataProjectors.sparkContext
+    dataProjectors
+      .leftOuterJoin(priorRandomEffectModelOpt.map(_.modelsRDD).getOrElse(sc.emptyRDD[(REId, GeneralizedLinearModel)]))
+      .mapValues { case (projector: LinearSubspaceProjector, priorModelOpt: Option[GeneralizedLinearModel]) =>
+        val activeCoef = priorModelOpt.map {
+            model =>
+              val means = model.coefficients.means
+              VectorUtils.getActiveIndices(means)
+        }.getOrElse(Set[Int]())
+        val projectedKeySet = projector.originalToProjectedSpaceMap.keySet
+
+        new LinearSubspaceProjector(activeCoef.union(projectedKeySet).toSet, originalSpaceDimension)
+      }
```

Comment:
There's a much cleaner way to write this function:
```
    val sc = SparkSession.builder().getOrCreate().sparkContext
    val originalSpaceDimension = keyedGameDataset
      .take(1)
      .head
      ._2
      ._2
      .features
      .length
    val priorModels = priorRandomEffectModelOpt.map(_.modelsRDD).getOrElse(sc.emptyRDD[(REId, GeneralizedLinearModel)])
    
    keyedGameDataset
      .leftOuterJoin(priorModels)
      .mapValues { case ((_, labeledPoint), priorModelOpt) =>
        val dataActiveIndices = VectorUtils.getActiveIndices(labeledPoint.features)
          
        priorModelOpt match {
          case Some(priorModel) => dataActiveIndices.union(VectorUtils.getActiveIndices(priorModel.coefficients.means))
          case None => dataActiveIndices
        }
      }
      .foldByKey(mutable.Set[Int](), randomEffectPartitioner)(_.union(_))
      .mapValues(activeIndices => new LinearSubspaceProjector(activeIndices.toSet, originalSpaceDimension))
```

Reply:
Agreed. It is much cleaner.